### ISSUE TITLE
Java REST runner interprets integers as doubles

### ISSUE THREAD

**Issue by clintongormley** [elasticsearch/elasticsearch#9931]
In https://github.com/elasticsearch/elasticsearch/blob/master/rest-api-spec/test/script/30_expressions.yaml#L28 the JSON response is:

    ...
              {
                 "fields" : {
                    "my_field" : [
                       23
                    ]
                 },
    ...

But the number `23` is interpreted by the Java REST runner as `23.0`, which means that the assertion that it should be `23` fails.

**Comment by clintongormley** [comment]
@javanna could you take a look at what can be done here please?  If it is not easily fixable, then we could change it to check for >= 23 and <= 23, but that feels suboptimal...

**Comment by javanna** [comment]
I had a look and I see something different. The json response that I print out as a string before any parsing is the following:

```
{took=340, timed_out=false, _shards={total=2, successful=2, failed=0}, hits={total=1, max_score=1.0, hits=[{_index=test123, _type=test, _id=1, _score=1.0, fields={my_field=[23.0]}}]}}
```

As far as I can see the expression lang is responsible for returning the integer field as a float. The java REST runner just uses the same parser that elasticsearch uses internally, which tries and detect the different numeric types and does see the difference between floats and integers.

I am afraid we have two potentials problems then:
- numbers are always float when using the expression language, maybe not a problem after all, we just need to be aware of this?
- the perl runner is somehow interpreting `23.0` as `23` and makes it an integer?

Long story short I don't see any problem with the java REST tests runner here... as far as I can see the original test that was checking for `23.0` was correct, cause that's what the expression script returns. Can you double check please @clintongormley ?

**Comment by clintongormley** [comment]
@javanna you are right - this is a Perl issue.  It silently converts `23.0` in the JSON into an integer (which is why it logged the integer in the trace logs).  Apologies for the noise - I'll probably have to work around this with a range check.
